msdk-id-sirius/src/main/java/SiriusIdentificationMethod.java

```diff
@@ -0,0 +1,349 @@
+    */
+  public static class ConstraintsGenerator {
+    private final String[] defaultElementSymbols = new String[]{"C", "H", "N", "O", "P"};
+    private final Element[] defaultElements;
+    private final PeriodicTable periodicTable = PeriodicTable.getInstance();
+    private final int maxNumberOfOneElements = 20;
+
+    /**
+     * <p>Constructor for ConstraintsGenerator class</p>
+     * Initializes array of Elements `defaultElements`
+     */
+    public ConstraintsGenerator() {
+      defaultElements = new Element[defaultElementSymbols.length];
+      for (int i = 0; i < defaultElementSymbols.length; i++)
+        defaultElements[i] = periodicTable.getByName(defaultElementSymbols[i]);
+    }
+
+    /**
+     * <p> Method for generating FormulaConstraints from user-defined search space</p>
+     * Parses isotopes from input parameter and transforms it into Element objects and sets their range value
+     * @param range - User defined search space of possible elements
+     * @return new Constraint to be used in Sirius
+     */
+    public FormulaConstraints generateConstraint(MolecularFormulaRange range) {
+      logger.debug("ConstraintsGenerator started procesing");
+      int size = range.getIsotopeCount();
+      Element elements[] = Arrays.copyOf(defaultElements, defaultElements.length + size);
+      int k = 0;
+
+      // Add items from `range` into array with default elements
+      for (IIsotope isotope: range.isotopes()) {
+        int atomicNumber = isotope.getAtomicNumber();
+        final Element element = periodicTable.get(atomicNumber);
+        elements[defaultElements.length + k++] = element;
+      }
+
+      // Generate initial constraint w/o concrete Element range
+      FormulaConstraints constraints = new FormulaConstraints(new ChemicalAlphabet(elements));
+
+      // Specify each Element range
+      for (IIsotope isotope: range.isotopes()) {
+        int atomicNumber = isotope.getAtomicNumber();
+        final Element element = periodicTable.get(atomicNumber);
+        int min = range.getIsotopeCountMin(isotope);
+        int max = range.getIsotopeCountMax(isotope);
+
+        constraints.setLowerbound(element, min);
+        if (max!= maxNumberOfOneElements) constraints.setUpperbound(element, max);
+      }
+
+      logger.debug("ConstraintsGenerator finished");
+      return constraints;
+    }
+
+  }
+
+  public double getParentMass() {
+    return parentMass;
+  }
+
+  public IonType getIonization() {
+    return ion;
+  }
+
+  public List<MsSpectrum> getMsSpectra() {
+    return ms1;
+  }
+
+  public List<MsSpectrum> getMs2Spectra() {
+    return ms2;
+  }
+
+  public int getNumberOfCandidates() {
+    return numberOfCandidates;
+  }
+
+  /**
+   *
+   * <p> Method for reading custom MS files (just 2 columns with mz & intentsity values </p>
+   * Does similar to mgf parser functionality
+   */
+  public static MsSpectrum readCustomMsFile(File file, String delimeter)
```

Comment:
This does not belong here. If you want to have an import function for such files, you should create a corresponding io module. But it seems to me this does the same as the MSP import module that is already there: https://github.com/msdk/msdk/blob/master/msdk-io-msp/src/main/java/io/github/msdk/io/msp/MspImportAlgorithm.java

Reply:
Well, it does quite similar stuff as the msp import module, but the files' structure differs.
This method does not have any specific structure.

This method allows to test on not-structured files.